test/MOI/MOI_wrapper.jl

```diff
@@ -1177,6 +1179,196 @@ function test_indicator_nonconstant()
+        MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([-2.0, -1.0], x), 0.0),
+        MOI.LessThan(-1.0),
+    )
+    MOI.optimize!(model)
+    @test MOI.get(model, MOI.TerminationStatus()) == MOI.INFEASIBLE
+    @test MOI.get(model, MOI.DualStatus()) == MOI.INFEASIBILITY_CERTIFICATE
+    clb_dual = MOI.get.(model, MOI.ConstraintDual(), clb)
+    c_dual = MOI.get(model, MOI.ConstraintDual(), c)
+    @show clb_dual, c_dual
+    @test clb_dual[1] < 1e-6
+    @test clb_dual[2] < 1e-6
+    @test c_dual[1] < 1e-6
+    @test clb_dual[1] ≈ 2 * c_dual atol = 1e-6
+    @test clb_dual[2] ≈ c_dual atol = 1e-6
+end
+
+function test_farkas_dual_max()
+    model = Gurobi.Optimizer(GRB_ENV)
+    MOI.set(model, MOI.Silent(), true)
+    MOI.set(model, MOI.RawParameter("InfUnbdInfo"), 1)
+    x = MOI.add_variables(model, 2)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    MOI.set(
+        model,
+        MOI.ObjectiveFunction{MOI.SingleVariable}(),
+        MOI.SingleVariable(x[1]),
+    )
+    clb = MOI.add_constraint.(
+        model, MOI.SingleVariable.(x), MOI.GreaterThan(0.0)
+    )
+    c = MOI.add_constraint(
+        model,
+        MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([2.0, 1.0], x), 0.0),
+        MOI.LessThan(-1.0),
+    )
+    MOI.optimize!(model)
+    @test MOI.get(model, MOI.TerminationStatus()) == MOI.INFEASIBLE
+    @test MOI.get(model, MOI.DualStatus()) == MOI.INFEASIBILITY_CERTIFICATE
+    clb_dual = MOI.get.(model, MOI.ConstraintDual(), clb)
+    c_dual = MOI.get(model, MOI.ConstraintDual(), c)
+    @show clb_dual, c_dual
+    @test clb_dual[1] > -1e-6
+    @test clb_dual[2] > -1e-6
+    @test c_dual[1] < 1e-6
+    @test clb_dual[1] ≈ -2 * c_dual atol = 1e-6
+    @test clb_dual[2] ≈ -c_dual atol = 1e-6
+end
+
+function test_farkas_dual_max_ii()
+    model = Gurobi.Optimizer(GRB_ENV)
+    MOI.set(model, MOI.Silent(), true)
+    MOI.set(model, MOI.RawParameter("InfUnbdInfo"), 1)
+    x = MOI.add_variables(model, 2)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    MOI.set(
+        model,
+        MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+        MOI.ScalarAffineFunction([MOI.ScalarAffineTerm(-1.0, x[1])], 0.0),
+    )
+    clb = MOI.add_constraint.(
+        model, MOI.SingleVariable.(x), MOI.LessThan(0.0)
+    )
+    c = MOI.add_constraint(
+        model,
+        MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([-2.0, -1.0], x), 0.0),
+        MOI.LessThan(-1.0),
+    )
+    MOI.optimize!(model)
+    @test MOI.get(model, MOI.TerminationStatus()) == MOI.INFEASIBLE
+    @test MOI.get(model, MOI.DualStatus()) == MOI.INFEASIBILITY_CERTIFICATE
+    clb_dual = MOI.get.(model, MOI.ConstraintDual(), clb)
+    c_dual = MOI.get(model, MOI.ConstraintDual(), c)
+    @show clb_dual, c_dual
+    @test clb_dual[1] < 1e-6
+    @test clb_dual[2] < 1e-6
+    @test_broken c_dual[1] < 1e-6
+    @test_broken clb_dual[1] ≈ 2 * c_dual atol = 1e-6
+    @test_broken clb_dual[2] ≈ c_dual atol = 1e-6
```

Comment:
TODO: Gurobi, including 9.1 is broken on these tests. I have a support request in.

Reply:
Here's an example demonstrating this is clearly a bug. The sign switches at a lower bound of `-1e20`, which is the value at which Gurobi ignores the lower bound.
```Julia
function test_farkas(obj_coefficient, lower_bound)
    env = Ref{Ptr{Cvoid}}()
    ret = GRBloadenv(env, C_NULL)
    GRBsetintparam(env[], "InfUnbdInfo", 1)
    GRBsetintparam(env[], "OutputFlag", 0)
    model = Ref{Ptr{Cvoid}}()
    GRBloadmodel(
        env[],                      # *env
        model,                      # **modelp
        C_NULL,                     # *Pname
        Cint(1),                    # numvars
        Cint(1),                    # numconstrs
        Cint(1),                    # objsense
        0.0,                        # objcon
        Cdouble[obj_coefficient],   # *obj
        Cchar[GRB_GREATER_EQUAL],   # *sense
        Cdouble[1.0],               # *rhs
        Cint[0],                    # *vbeg
        Cint[1],                    # vlen
        Cint[0],                    # *vind
        Cdouble[1.0],               # *vval
        Cdouble[lower_bound],       # *lb
        Cdouble[0.0],               # *ub
        C_NULL,                     # *vtype
        C_NULL,                     # **varnames
        C_NULL,                     # **constrnames
    )
    GRBoptimize(model[])
    dual = Vector{Cdouble}(undef, 1)
    GRBgetdblattrarray(model[], "FarkasDual", 0, 1, dual)
    return dual[1]
end

julia> test_farkas(1.0, -1e20)
Academic license - for non-commercial use only - expires 2020-12-05
1.0

julia> test_farkas(1.0, nextfloat(-1e20))
Academic license - for non-commercial use only - expires 2020-12-05
-1.0

julia> a, b, c = Ref{Cint}(), Ref{Cint}(), Ref{Cint}()
(Base.RefValue{Int32}(482535088), Base.RefValue{Int32}(482535088), Base.RefValue{Int32}(482535152))

julia> GRBversion(a, b, c)

julia> a, b, c
(Base.RefValue{Int32}(9), Base.RefValue{Int32}(1), Base.RefValue{Int32}(0))
```

Reply:
Still waiting for Gurobi support, but I'm now fairly confident that this is a bug in Gurobi, and therefore not blocking for this PR. 

If they fix it in a future release, the `@test_broken` will break (pass) and we can address.

Reply:
Bug confirmed upstream. Will be fixed in next release.